splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1331,6 +1333,37 @@ public void removeFKDependencyOnPrivileges(TransactionController tc) throws Stan
                 "SYS.SYSDEPENDS updated: Foreign keys dependencies on RoleDescriptors or permission descriptors deleted, total rows deleted: " + rowsToDelete.size());
     }
 
+    public int upgradeTablePriorities(TransactionController tc) throws Exception {
+        PartitionAdmin admin = SIDriver.driver().getTableFactory().getAdmin();
+        ArrayList<String> toUpgrade = new ArrayList<>();
+        Function<TabInfoImpl, Void> addTabInfo =  (TabInfoImpl info ) ->
+                {
+                    toUpgrade.add( Long.toString(info.getHeapConglomerate()) );
+                    for( int j = 0; j < info.getNumberOfIndexes(); j++ )
+                        toUpgrade.add( Long.toString(info.getIndexConglomerate(j)) );
+                    return null;
+                };
+        for (int i = 0; i < coreInfo.length; ++i) {
+            assert coreInfo[i] != null;
+            addTabInfo.apply(coreInfo[i]);
+        }
+        for (int i = 0; i < NUM_NONCORE; ++i) {
+            // noncoreInfo[x] will be null otherwise
+            TabInfoImpl ti = getNonCoreTIByNumber(i+NUM_CORE);
```

Comment:
Nitpick: use getNonCoreTI(), which invokes faultInTabInfo for you.

Reply:
done